### PR TITLE
Add additional resolver tests

### PR DESCRIPTION
**Pull request by rigtorp**
**Changes**

Add tests to make sure the TryClean bug is never reintroduced
Match years in *arabic numerals only* delimited by any unicode non-word characters

**Issues**

### CODE REVIEW

File: Emby.Naming/Common/NamingOptions.cs [hunk 1]
```diff
@@ -137,7 +137,9 @@ public NamingOptions()
             CleanDateTimes = new[]
             {
                 @"(.+[^_\,\.\(\)\[\]\-])[_\.\(\)\[\]\-](19\d{2}|20\d{2})([ _\,\.\(\)\[\]\-][^0-9]|).*(19\d{2}|20\d{2})*",
-                @"(.+[^_\,\.\(\)\[\]\-])[ _\.\(\)\[\]\-]+(19\d{2}|20\d{2})([ _\,\.\(\)\[\]\-][^0-9]|).*(19\d{2}|20\d{2})*"
+                @"(.+[^_\,\.\(\)\[\]\-])[ _\.\(\)\[\]\-]+(19\d{2}|20\d{2})([ _\,\.\(\)\[\]\-][^0-9]|).*(19\d{2}|20\d{2})*",
+                @"(.+\w)\W+\p{Ps}(19[0-9]{2}|20[0-9]{2})\p{Pe}", // Prefer year enclosed in parenthesis (){}[]
+                @"(.+\w)\W+(19[0-9]{2}|20[0-9]{2})(\W|$)", // Secondarily year surounded by non-word chars
```
Comment (Artiume) [inline Emby.Naming/Common/NamingOptions.cs:142]: secondary year surrounded

File: tests/Jellyfin.Naming.Tests/Video/VideoResolverTests.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using System.Collections.Generic;
+﻿﻿using System.Collections.Generic;
```
Comment (Artiume) [inline tests/Jellyfin.Naming.Tests/Video/VideoResolverTests.cs:1]: was the BOM changed?

Comment (rigtorp) [reply tests/Jellyfin.Naming.Tests/Video/VideoResolverTests.cs]: Yes, a mistake.

Comment (rigtorp) [reply tests/Jellyfin.Naming.Tests/Video/VideoResolverTests.cs]: I fixed the BOM issue.

**Comment by joshuaboniface** [comment]
@jellyfin/server This one has fallen through the cracks, can we get feedback?

**Review by BaronGreenback — APPROVED** [review]

**Review by EraYaN — APPROVED** [review]

**Comment by EraYaN** [comment]
Seems to be an syntax error: 

```
Common/NamingOptions.cs(140,166): error CS1003: Syntax error, ',' expected
```

File: Emby.Naming/Common/NamingOptions.cs [hunk 3]
```diff
@@ -138,6 +138,8 @@ public NamingOptions()
             {
                 @"(.+[^_\,\.\(\)\[\]\-])[_\.\(\)\[\]\-](19[0-9]{2}|20[0-9]{2})(?![0-9]+|\W[0-9]{2}\W[0-9]{2})([ _\,\.\(\)\[\]\-][^0-9]|).*(19[0-9]{2}|20[0-9]{2})*",
                 @"(.+[^_\,\.\(\)\[\]\-])[ _\.\(\)\[\]\-]+(19[0-9]{2}|20[0-9]{2})(?![0-9]+|\W[0-9]{2}\W[0-9]{2})([ _\,\.\(\)\[\]\-][^0-9]|).*(19[0-9]{2}|20[0-9]{2})*"
```
Comment (BaronGreenback) [inline Emby.Naming/Common/NamingOptions.cs:140]:
```suggestion
                @"(.+[^_\,\.\(\)\[\]\-])[ _\.\(\)\[\]\-]+(19[0-9]{2}|20[0-9]{2})(?![0-9]+|\W[0-9]{2}\W[0-9]{2})([ _\,\.\(\)\[\]\-][^0-9]|).*(19[0-9]{2}|20[0-9]{2})*",
```

**Review by crobibero — APPROVED** [review]

**Pull request merged**